air/src/stack/stack_flag/mod.rs

```diff
@@ -0,0 +1,801 @@
+
+    #[allow(dead_code)]
+    fn rpperm_op_flag(&self) -> E {
+        self.degree6_op_flags[8]
+    }
+
+    #[allow(dead_code)]
+    fn mpverify_op_flag(&self) -> E {
+        self.degree6_op_flags[9]
+    }
+
+    #[allow(dead_code)]
+    fn span_op_flag(&self) -> E {
+        self.degree6_op_flags[12]
+    }
+
+    #[allow(dead_code)]
+    fn join_op_flag(&self) -> E {
+        self.degree6_op_flags[13]
+    }
+
+    #[allow(dead_code)]
+    fn split_op_flag(&self) -> E {
+        self.degree6_op_flags[14]
+    }
+
+    #[allow(dead_code)]
+    fn loop_op_flag(&self) -> E {
+        self.degree6_op_flags[15]
+    }
+
+    // ------ Degree 4 stack operations  -----------------------------------------------------
+
+    #[allow(dead_code)]
+    fn mrupdate_op_flag(&self) -> E {
+        self.degree4_op_flags[0]
+    }
+
+    #[allow(dead_code)]
+    fn push_op_flag(&self) -> E {
+        self.degree4_op_flags[1]
+    }
+
+    #[allow(dead_code)]
+    fn end_op_flag(&self) -> E {
+        self.degree4_op_flags[4]
+    }
+
+    #[allow(dead_code)]
+    fn repeat_op_flag(&self) -> E {
+        self.degree4_op_flags[5]
+    }
+
+    #[allow(dead_code)]
+    fn respan_op_flag(&self) -> E {
+        self.degree4_op_flags[6]
+    }
+
+    #[allow(dead_code)]
+    fn halt_op_flag(&self) -> E {
+        self.degree4_op_flags[7]
+    }
+}
+
+trait EvaluationFrameExt<E: FieldElement> {
+    // --- Operation bit accessors ---------------------------------------------------------------
+
+    /// Returns the current value of the specified operation bit in the decoder. It assumes that
+    /// the index is a valid index.
+    fn op_bit_item(&self, index: usize) -> E;
+
+    /// Returns the value of operation bit extra column which is used to reduce the degree of op
+    /// flags for degree 4 operations where the two most significant bits are ONE.
+    fn helper_register(&self) -> E;
+}
+
+impl<E: FieldElement> EvaluationFrameExt<E> for &EvaluationFrame<E> {
+    // --- Operation bit accessors ---------------------------------------------------------------
+
+    #[inline]
+    fn op_bit_item(&self, idx: usize) -> E {
+        self.current()[DECODER_TRACE_OFFSET + OP_BITS_OFFSET + idx]
+    }
+
+    #[inline]
+    fn helper_register(&self) -> E {
+        self.current()[DECODER_TRACE_OFFSET + OP_BIT_EXTRA_COL_IDX]
+    }
+}
+
+// HELPER FUNCTIONS
+// ------------------------------------------------------------------------------------------------
+
+/// This helper function generates all the flag values of degree seven stack operations.
+/// This degree seven stack operation includes operations that shift the stack to the right,
+/// and left and don't shift at all.
+fn set_op_flags_degree_7<E: FieldElement>(frame: &EvaluationFrame<E>) -> [E; NUM_DEGREE_7_OPS] {
+    let mut degree_7_intermediate_values = [E::ZERO; NUM_DEGREE_7_OPS];
+
+    let not_6 = binary_not(frame.op_bit_item(6));
+    let not_5 = binary_not(frame.op_bit_item(5));
```

Comment:
In the description of the decoder, especially in the operational bits part, the least significant bit is b6, whereas, in the case of the operation bit description in the operand stack, it's the opposite, i.e., b0 is the least significant bit. Although it's not a big deal but we should keep the notation consistent in the docs and code. I've for the time being went with decoder notation.

Reply:
I've updated the logic accordingly. I will fix it in the docs in a separate PR.